ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -378,6 +383,31 @@ short getVersion() {
     return toReturn != null ? Collections.unmodifiableNavigableSet(toReturn) : null;
   }
 
+  /**
+   * Generate bloom file by walking through all index entries in this segment and persist it.
+   * @throws IOException
+   */
+  private void generateBloomFileAndPersist() throws IOException {
+    List<IndexEntry> entries = new ArrayList<>();
+    getIndexEntriesSince(null, new FindEntriesCondition(Long.MAX_VALUE), entries, new AtomicLong(0), true);
+    for (IndexEntry entry : entries) {
```

Comment:
just before this is where we should instantiate the bloom filter since we will now know the size of the filter

Reply:
done. Thanks for suggestion.